### ISSUE TITLE
Check backend should softmax policy before calculating errors

### ISSUE THREAD

**Issue by Tilps** [LeelaChessZero/lc0#1263]
Since softmax was originally done by backends when check backend was written - any ideas about what is acceptable policy absolute/relative errors no longer apply since it switched to receiving raw logits.
Since softmax output is identical for all fixed offsets of logits, it is possible to have an absolute error for all policies that is large, but for it to also not matter at all.

Unfortunately check backend doesn't have the legal move list to go with the data - so its not clear how it can calculate the 'correct' softmax, since network output for illegal moves can be anything due to training with masking and that makes softmax over all outputs not a useful thing

**Comment by borg323** [comment]
Looks like to fix this we'll have to change the backend interface.
